src/Penrose/Functions.hs

```diff
@@ -492,58 +489,48 @@ checkReturn (GPI v) _ = error "checkReturn: Computations cannot return GPIs"
 --------------------------------------------------------------------------------
 -- Computation Functions
 sampleFunction :: CompFn
--- Assuming domain and range are lines or arrows, TODO deal w/ points
--- TODO: discontinuous functions? not sure how to sample/model/draw consistently
-sampleFunction [Val (IntV n), GPI domain, GPI range] g =
-  let (dsx, dsy, dex, dey) =
-        ( getNum domain "startX"
-        , getNum domain "startY"
-        , getNum domain "endX"
-        , getNum domain "endY")
-      (rsx, rsy, rex, rey) =
-        ( getNum range "startX"
-        , getNum range "startY"
-        , getNum range "endX"
-        , getNum range "endY")
-      lower_left = (min dsx dex, min rsy rey)
-      top_right = (max dsx dex, max rsy rey)
-      (pts, g') = computeSurjection g n lower_left top_right
-  in (Val $ PtListV pts, g')
-
--- Computes the surjection to lie inside a bounding box defined by the corners of a box
--- defined by four straight lines, assuming their lower/left coordinates come first.
--- Their intersections give the corners.
-computeSurjectionLines :: CompFn
-computeSurjectionLines args g =
-  let (pts, g') = computeSurjectionLines' g args
-  in (Val $ PtListV pts, g')
-
-computeSurjectionLines' ::
-     (Autofloat a) => StdGen -> [ArgVal a] -> ([Pt2 a], StdGen)
-computeSurjectionLines' g args@[Val (IntV n), GPI left@("Line", _), GPI right@("Line", _), GPI bottom@("Line", _), GPI top@("Line", _)] =
-  let lower_left = (getNum left "startX", getNum bottom "startY")
-  in let top_right = (getNum right "startX", getNum top "startY")
-     in computeSurjection g n lower_left top_right
--- Assuming left and bottom are perpendicular and share one point
-computeSurjectionLines' g [Val (IntV n), GPI left@("Arrow", _), GPI bottom@("Arrow", _)] =
-  let lower_left = (getNum left "startX", getNum left "startY")
-  in let top_right = (getNum bottom "endX", getNum left "endY")
-     in computeSurjection g n lower_left top_right
-
-computeSurjection ::
-     Autofloat a => StdGen -> Integer -> Pt2 a -> Pt2 a -> ([Pt2 a], StdGen)
+sampleFunction [Val (IntV n), Val (FloatV xmin), Val (FloatV xmax), Val (FloatV ymin), Val (FloatV ymax), Val (StrV typ)] g 
```

Comment:
I think it might be marginally faster to have these as separate functions (sampleSurjection, etc) so the runtime doesn't need to fall thru the case statement every time the function is evaluated. But it's probably not that much faster.

Reply:
I'm not sure how the `Map` data structure in Haskell works. If the lookup cost scales with the size of the dict, there might be a trade-off between the two. Well, I'm personally inclined to have fewer more general comp functions. Let me know what you think.